### PR TITLE
Copy the glibc bug fix to devicelab/integration tests

### LINKED ISSUE #70493: Flake in web integration test shard
```
▌21:20:45▐ ELAPSED TIME: 8.491s for ../../../bin/flutter build web --release --sound-null-safety -t lib/sound_mode.dart in dev/integration_tests/web
Launching Chromium 84.0.4147.0 

[CHROME]: Fontconfig warning: "/etc/fonts/fonts.conf", line 100: unknown element "blank"
[CHROME]: Inconsistency detected by ld.so: ../elf/dl-tls.c: 493: _dl_allocate_tls_init: Assertion `listp->slotinfo[cnt].gen <= GL(dl_tls_generation)' failed!
Unhandled exception:
Exception: Expected Chrome to print "DevTools listening" string with DevTools URL, but the string was never printed.
#0      _connectToChromeDebugPort.<anonymous closure> (package:flutter_devicelab/framework/browser.dart:262:7)
#1      _runUserCode (dart:async/stream_pipe.dart:11:23)
#2      Stream.firstWhere.<anonymous closure> (dart:async/stream.dart:1414:9)
#3      _RootZone.runGuarded (dart:async/zone.dart:1368:10)
#4      _BufferingStreamSubscription._sendDone.sendDone (dart:async/stream_impl.dart:410:13)
#5      _BufferingStreamSubscription._sendDone (dart:async/stream_impl.dart:420:15)
#6      _BufferingStreamSubscription._close (dart:async/stream_impl.dart:305:7)
#7      _ForwardingStream._handleDone (dart:async/stream_pipe.dart:101:10)
#8      _ForwardingStreamSubscription._handleDone (dart:async/stream_pipe.dart:165:13)
#9      _RootZone.runGuarded (dart:async/zone.dart:1368:10)
#10     _BufferingStreamSubscription._sendDone.sendDone (dart:async/stream_impl.dart:410:13)
#11     _BufferingStreamSubscription._sendDone (dart:async/stream_impl.dart:420:15)
#12     _BufferingStreamSubscription._close (dart:async/stream_impl.dart:305:7)
#13     _SinkTransformerStreamSubscription._close (dart:async/stream_transformers.dart:97:11)
#14     _EventSinkWrapper.close (dart:async/stream_transformers.dart:23:11)
#15     _StringAdapterSink.close (dart:convert/string_conversion.dart:251:11)
#16     _LineSplitterSink.close (dart:convert/line_splitter.dart:139:11)
#17     _SinkTransformerStreamSubscription._handleDone (dart:async/stream_transformers.dart:142:24)
#18     _RootZone.runGuarded (dart:async/zone.dart:1368:10)
#19     _BufferingStreamSubscription._sendDone.sendDone (dart:async/stream_impl.dart:410:13)
#20     _BufferingStreamSubscription._sendDone (dart:async/stream_impl.dart:420:15)
#21     _BufferingStreamSubscription._close (dart:async/stream_impl.dart:305:7)
#22     _SinkTransformerStreamSubscription._close (dart:async/stream_transformers.dart:97:11)
#23     _EventSinkWrapper.close (dart:async/stream_transformers.dart:23:11)
#24     _StringAdapterSink.close (dart:convert/string_conversion.dart:251:11)
#25     _Utf8ConversionSink.close (dart:convert/string_conversion.dart:305:20)
#26     _ConverterStreamEventSink.close (dart:convert/chunked_conversion.dart:81:18)
#27     _SinkTransformerStreamSubscription._handleDone (dart:async/stream_transformers.dart:142:24)
#28     _RootZone.runGuarded (dart:async/zone.dart:1368:10)
#29     _BufferingStreamSubscription._sendDone.sendDone (dart:async/stream_impl.dart:410:13)
#30     _BufferingStreamSubscription._sendDone (dart:async/stream_impl.dart:420:15)
#31     _BufferingStreamSubscription._close (dart:async/stream_impl.dart:305:7)
#32     _SyncStreamControllerDispatch._sendDone (dart:async/stream_controller.dart:815:19)
#33     _StreamController._closeUnchecked (dart:async/stream_controller.dart:670:7)
#34     _StreamController.close (dart:async/stream_controller.dart:663:5)
#35     _Socket._onData (dart:io-patch/socket_patch.dart:1976:21)
#36     _RootZone.runUnaryGuarded (dart:async/zone.dart:1380:10)
#37     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:357:11)
#38     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:285:7)
#39     _SyncStreamControllerDispatch._sendData (dart:async/stream_controller.dart:807:19)
#40     _StreamController._add (dart:async/stream_controller.dart:681:7)
#41     _StreamController.add (dart:async/stream_controller.dart:624:5)
#42     new _RawSocket.<anonymous closure> (dart:io-patch/socket_patch.dart:1512:35)
#43     _NativeSocket.issueReadEvent.issue (dart:io-patch/socket_patch.dart:1010:18)
#44     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
#45     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
#46     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:120:13)
#47     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:185:5)
```

https://ci.chromium.org/p/flutter/builders/prod/Linux%20web_integration_tests/1025

### PR DESCRIPTION
**Pull request by yjbanov**
## Description

Handle the glibc bug that Chrome occasionally hits in the integration/devicelab tests.

This is just copy'n'paste of https://github.com/flutter/flutter/pull/67466 into the devicelab, which doesn't share code with Flutter tool.

## Related Issues

Fixes https://github.com/flutter/flutter/issues/70493

### CODE REVIEW

**Review by jonahwilliams — APPROVED** [review]
LGTM

File: dev/devicelab/lib/framework/browser.dart [hunk 1]
```diff
@@ -576,3 +558,72 @@ int _readInt(Map<String, dynamic> json, String key) {
...
+/// Used by [Chrome.launch] to detect a glibc bug and retry launching the
+/// browser.
+///
+/// Once every few thousands of launches we hit this glibc bug:
+///
+/// https://sourceware.org/bugzilla/show_bug.cgi?id=19329.
+///
+/// When this happens Chrome spits out something like the following then exits with code 127:
+///
+///     Inconsistency detected by ld.so: ../elf/dl-tls.c: 493: _dl_allocate_tls_init: Assertion `listp->slotinfo[cnt].gen <= GL(dl_tls_generation)' failed!
```
Comment (nturgut) [inline dev/devicelab/lib/framework/browser.dart:571]: very nit: is this spaces intentional?

**Review by nturgut — APPROVED** [review]
LGTM.

I think we have one more chrome left, the one in the felt code.

I only saw it creating this error a few times. It might be good to create a P5 issue for also doing the same thing for that one.

File: dev/devicelab/lib/framework/browser.dart (hunk 1 above)
Comment (yjbanov) [reply dev/devicelab/lib/framework/browser.dart:571]: Yes, the 4-space indent is treated by dartdoc as a code snippet.

Comment (nturgut) [reply dev/devicelab/lib/framework/browser.dart:571]: thanks! I haven't used this before.

**Comment by yjbanov** [comment]
@nturgut 

> It might be good to create a P5 issue for also doing the same thing for that one.

Good point: https://github.com/flutter/engine/pull/22878

**Pull request merged**